File: 2750.c
#include <stdio.h>

void swap(int* a, int* b)
{
    int tmp = *a;
    *a = *b;
    *b = tmp;
}

void quicksort(int arr[], int left, int right)
{
    int pivot = arr[(left + right)/2];
    int i = left;
    int j = right;

    while (i <= j){

        while (arr[i] < pivot){i++;}

        while (arr[j] > pivot){j--;}

        if (i <= j){
            //tmp = arr[i];
            //arr[i] = arr[j];
            //arr[j] = tmp;
            swap(&arr[i],&arr[j]);
            i++;
            j--;
        }
    }
    if (left < j)
        quicksort(arr, left, j);
    if (i < right)
        quicksort(arr, i, right);
}

void merge(int arr[], int start, int mid, int end)
{
    int length = end - start + 1;
    int temp[length];

    int left = start;
    int right = mid+1;
    int i;

    for (i=0; i<length; i++){
        if (left <= mid && right <= end){
            if (arr[left] < arr[right]){
                temp[i] = arr[left];
                left++;
            }
            else {
                temp[i] = arr[right];
                right++;
            }
        }

        else if (left > mid && right <= end){
            temp[i] = arr[right];
            right++;
        }
        else if (left <= mid && right > end){
            temp[i] = arr[left];
            left++;
        }
    }
    for (i=start; i<=end; i++){
        arr[i] = temp[i-start];
    }
}

void mergesort(int arr[], int start, int end)
{
    if (start >= end)
        return;
    
    int mid = (start + end) / 2;

    mergesort(arr,start,mid);
    mergesort(arr,mid+1,end);

    merge(arr,start,mid,end);

}
int main(void)
{
    int i, n;
    scanf("%d",&n);

    int arr[n];
    for (i=0;i<n;i++)
    {
        scanf("%d",&arr[i]);
    }

    //quicksort(arr,0,n-1);
    mergesort(arr,0,n-1);
    for (i=0;i<n;i++)
    {
        printf("%d\n",arr[i]);
    }
    
    return 0;
}

File: 2750.py
def quicksort(s):
    if len(s) == 0:
        return [] 
    
    elif len(s) == 1:
        return s

    else:
        pivot = s[len(s)/2]
        left = []
        right = []

        for e in s:
            if e < pivot:
                left.append(e)
            elif e > pivot:
                right.append(e)
        return quicksort(left) + [pivot] + quicksort(right)

def merge(left, right):
    n = len(left+right)
    mergedList = []
    while len(mergedList) < n:
        if len(left) == 0:
            mergedList.append(right.pop(0))
        elif len(right) == 0:
            mergedList.append(left.pop(0))
        else:
            mergedList.append(left.pop(0) if left[0] < right[0] else right.pop(0))

    return mergedList

def mergesort(s):
    if len(s) == 0:
        return []
    elif len(s) == 1:
        return s
    else:
        left = s[:len(s)/2]
        right = s[len(s)/2:]
        
        return merge(mergesort(left),mergesort(right))




lines = input()
tempList = []
for i in range(lines):
    tempList.append(input())

#print '\n'.join(str(e) for e in quicksort(tempList))
print '\n'.join(str(e) for e in mergesort(tempList))
